internal/nimparser/nimparser.go

```diff
@@ -127,75 +127,92 @@ func MatchProfiles(modelSpec appsv1alpha1.ModelSpec, manifest NIMManifest, disco
 	var selectedProfiles []string
 
 	for hash, profile := range manifest {
-		if modelSpec.Precision != "" && profile.Tags["precision"] != modelSpec.Precision {
-			// continue with the next profile in the manifest
+		// Check precision, tensor parallelism, and QoS profile
+		if (modelSpec.Precision != "" && profile.Tags["precision"] != modelSpec.Precision) ||
+			(modelSpec.TensorParallelism != "" && profile.Tags["tp"] != modelSpec.TensorParallelism) ||
+			(modelSpec.QoSProfile != "" && profile.Tags["profile"] != modelSpec.QoSProfile) {
 			continue
 		}
-		if modelSpec.TensorParallelism != "" && profile.Tags["tp"] != modelSpec.TensorParallelism {
-			// continue with the next profile in the manifest
-			continue
-		}
-		if modelSpec.QoSProfile != "" && profile.Tags["profile"] != modelSpec.QoSProfile {
-			// continue with the next profile in the manifest
+
+		// Check LoRA configuration
+		if modelSpec.Lora == nil && profile.Tags["feat_lora"] == "true" {
 			continue
 		}
 		if modelSpec.Lora != nil && profile.Tags["feat_lora"] != strconv.FormatBool(*modelSpec.Lora) {
-			// continue with the next profile in the manifest
 			continue
 		}
-		if modelSpec.Engine != "" && profile.Tags["llm_engine"] != modelSpec.Engine {
-			// continue with the next profile in the manifest
+
+		// Determine backend type
+		backend := profile.Tags["llm_engine"]
+		if backend == "" {
+			backend = profile.Tags["backend"]
+		}
+
+		if modelSpec.Engine != "" && !strings.Contains(backend, strings.TrimSuffix(modelSpec.Engine, "_llm")) {
```

Comment:
minor, but can we use == instead of Contains?

Reply:
@slu2011 i have used substring match just in case if the backend value turns out be something like `tensorrt_xxx` in the profile. Basically all these are optimized profiles we can filter.